protocol/src/traits/executor.rs

```diff
@@ -26,6 +28,12 @@ pub trait Executor: Send + Sync {
         data: Vec<u8>,
     ) -> TxResp;
 
+    fn cache_code<B: Backend + ApplyBackend + ExecutorAdapter>(
```

Comment:
This function will not be called by other modules, so it should not occur in the trait.